### ISSUE TITLE
backend should raise warning if circuit has no measure

### ISSUE THREAD

**Issue by ryan-woo** [Qiskit/qiskit-ibmq-provider#121]
### What is the expected enhancement?
The assembler should ensure that each circuit has a `measure` at the end before submission and raise an error if the circuit does not.

This would save some time where an invalid job would wait through a queue before it is determined to be invalid on the backend.

**Comment by jwoehr** [comment]
Am I correct that to accomplish this, one has to peel apart the `Qobj` passed to `IBMQBackend.run()` and parse each `qobjexp` for its `instructions` ?

**Comment by jwoehr** [comment]
Hmm, looking at this, it seems there should be a function in utils.py `warns_all_no_measures(qobj)` that can optionally be called from any backend and will toss a warning for every experiment in the `Qobj` which has no `measure`.

**Comment by diego-plan9** [comment]
Yes, that's a possibility, although let's regroup and reevaluate this issue and #125 after the next release (which should be ready soon-ish), taking a fresh look after the refactors have been completed. 

There is another component involved in the process: the API since a few months ago performs a validation of the qobj before sending it to the queue, which might make sense to use to alleviate the initial argument for the feature (`... save some time where an invalid job would wait through a queue before it is determined to be invalid on the backend`).
